tests/bsim/bluetooth/host/privacy/peripheral/src/dut.c

```diff
@@ -14,38 +14,97 @@
 #include <zephyr/bluetooth/conn.h>
 #include <zephyr/toolchain/gcc.h>
 
+#include "common/bt_str.h"
+
+#define ADV_SET_INDEX_ONE   0x00
+#define ADV_SET_INDEX_TWO   0x01
+#define ADV_SET_INDEX_THREE 0x02
+
+static struct bt_le_ext_adv *adv_set[CONFIG_BT_EXT_ADV_MAX_ADV_SET];
+
+static const struct bt_data ad_id[] = {
+	BT_DATA_BYTES(BT_DATA_MANUFACTURER_DATA, ADV_SET_INDEX_ONE),
+	BT_DATA_BYTES(BT_DATA_MANUFACTURER_DATA, ADV_SET_INDEX_TWO),
+	BT_DATA_BYTES(BT_DATA_MANUFACTURER_DATA, ADV_SET_INDEX_THREE),
+};
+
 bool cb_rpa_expired(struct bt_le_ext_adv *adv)
 {
 	/* Return true to rotate the current RPA */
+	int err;
+	struct bt_le_ext_adv_info info;
+
+	err = bt_le_ext_adv_get_info(adv, &info);
+	if (err) {
+		return false;
+	}
+	printk("advertiser[%d] RPA %s\n", info.id, bt_addr_le_str(info.addr));
 	return true;
 }
 
-void start_advertising(void)
+static void create_adv(struct bt_le_ext_adv **adv, int id)
 {
 	int err;
+	struct bt_le_adv_param params;
 	static struct bt_le_ext_adv_cb cb_adv;
-	struct bt_le_ext_adv_start_param start_params;
-	struct bt_le_ext_adv *adv;
+
+	cb_adv.rpa_expired = cb_rpa_expired;
+	memset(&params, 0, sizeof(struct bt_le_adv_param));
+
+	params.options |= BT_LE_ADV_OPT_EXT_ADV;
+	params.id = id;
+	params.sid = 0;
+	params.interval_min = BT_GAP_ADV_FAST_INT_MIN_1;
+	params.interval_max = BT_GAP_ADV_FAST_INT_MAX_1;
+
+	err = bt_le_ext_adv_create(&params, &cb_adv, adv);
+	if (err) {
+		FAIL("Failed to create advertiser (%d)\n", err);
+	}
+}
+
+void start_advertising(void)
+{
+	int err;
+	int id_a;
+	int id_b;
 
 	/* Enable bluetooth */
 	err = bt_enable(NULL);
 	if (err) {
 		FAIL("Failed to enable bluetooth (err %d\n)", err);
 	}
 
-	/* Start non-connectable extended advertising with private address */
-	start_params.timeout = 0;
-	start_params.num_events = 0;
-	cb_adv.rpa_expired = cb_rpa_expired;
+	id_a = bt_id_create(NULL, NULL);
 
-	err = bt_le_ext_adv_create(BT_LE_EXT_ADV_CONN_NAME, &cb_adv, &adv);
-	if (err) {
-		FAIL("Failed to create advertising set (err %d)\n", err);
-	}
+	ASSERT(id_a >= 0, "bt_id_create id_a failed (err %d)\n", id_a);
 
-	err = bt_le_ext_adv_start(adv, &start_params);
-	if (err) {
-		FAIL("Failed to enable periodic advertising (err %d)\n", err);
+	id_b = bt_id_create(NULL, NULL);
+
+	ASSERT(id_b >= 0, "bt_id_create id_b failed (err %d)\n", id_b);
```

Comment:
Suggest to use `FAIL` instead of `ASSERT`

Reply:
done.